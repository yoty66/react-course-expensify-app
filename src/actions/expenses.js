import uuid from "uuid";

//AddEXPENSE
export const addExpense = (
    {
        description = '',
        note = '' ,
        amount = 0 ,
        createdAt = 0,
    }={})=> ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }

});

export const removeExpense = ({id})=>
    ({
        type:'REMOVE_EXPENSE',
        id
    });

//editExpense
export const editExpense = (id ,edit)=>
    ({
        type:'EDIT_EXPENSE',
        id,
        edit
    });


