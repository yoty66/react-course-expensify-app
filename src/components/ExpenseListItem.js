import React from 'react';
import {NavLink} from "react-router-dom";
import moment from 'moment';
import numeral from 'numeral';


const ExpenseListItem = ({id,description, amount , createdAt }) => (
    <div>
        <NavLink to={`/edit/${id}`} > <h3>{description}</h3></NavLink>
        <p>
            {numeral(amount/100).format('$0,0.00').replace('$','â‚ª')}
            -
            {moment(createdAt).format('MMMM Do, YYYY')}
        </p>

    </div>
);

export default ExpenseListItem;