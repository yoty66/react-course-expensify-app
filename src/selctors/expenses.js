import moment from 'moment';

export const getVisableExpenses= ( expenses ,{startDate, endDate ,sortBy ,text } ) => {
    return expenses.filter (
        (expense) => {
            const createdAtMoment = moment(expense.createdAt);
            const startDateMatch =  startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true ;
            const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true ;
            const textMatch=expense.description.toLocaleLowerCase().includes(text.toLocaleLowerCase());
            return startDateMatch && endDateMatch && textMatch;
        }
    ).sort(
        (a,b) => {
            if (sortBy === 'date')
                return a.createdAt < b.createdAt ? 1: -1;
            else if (sortBy === 'amount' )
                return a.amount < b.amount ? 1 : -1 ;
        } );
};
