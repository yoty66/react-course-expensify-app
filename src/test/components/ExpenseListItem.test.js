import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixturs/expenses'

test('ExpenseListItem to render correctly ', ()=> {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});
