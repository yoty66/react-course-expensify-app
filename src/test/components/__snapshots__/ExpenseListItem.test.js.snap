// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseListItem to render correctly  1`] = `
<div>
  <NavLink
    to="/edit/1"
  >
     
    <h3>
      Gum
    </h3>
  </NavLink>
  <p>
    â‚ª1.95
    -
    January 1st, 1970
  </p>
</div>
`;
